packages/rocketchat-api/server/api.js

```diff
@@ -178,17 +178,26 @@ const getUserAuth = function _getUserAuth() {
 	};
 };
 
-RocketChat.API.v1 = new API({
-	version: 'v1',
-	useDefaultAuth: true,
-	prettyJson: true,
-	enableCors: false,
-	auth: getUserAuth()
-});
+let enableCors = RocketChat.settings.get('API_Enable_CORS');
+
+const createApi = function() {
+	RocketChat.API.v1 = new API({
+		version: 'v1',
+		useDefaultAuth: true,
+		prettyJson: true,
+		enableCors,
+		auth: getUserAuth()
+	});
+
+	RocketChat.API.default = new API({
+		useDefaultAuth: true,
+		prettyJson: true,
+		enableCors,
+		auth: getUserAuth()
+	});
+};
 
-RocketChat.API.default = new API({
-	useDefaultAuth: true,
-	prettyJson: true,
-	enableCors: false,
-	auth: getUserAuth()
+RocketChat.settings.onload('API_Enable_CORS', (key, value) => {
+	enableCors = value;
+	createApi();
```

Comment:
@graywolf336 that's what I meant writing y'days comment. 
Is that the proper way to do it?

_Note to myself: I really should get an IDE on my mobile..._

Reply:
You're extremely close! Honestly, I would leave the function the way it is and just change the `RocketChat.settings.onload` handler to be `RocketChat.settings.get('API_Enabled_CORS', () => {})` as that will be called everytime the setting is changed. I have no clue what `onload` will do :thinking:

Reply:
So, after doing some review it seems that `get` with a callback makes a call to `onload` and also immediately runs the callback..someone feel free to correct me if I'm misunderstanding things

Reply:
Is get() really being triggered after update of the setting? onLoad() surely is

Reply:
@rodrigok which one is the preferred way to do what is being discussed here?

Reply:
@graywolf336 Yes, `get` with `callback` will be called everytime the setting changes